### ISSUE TITLE
onMouseDown causes splash on real ios browsers

### ISSUE THREAD

**Issue by huhaixiao** [facebook/react#12717]
**Do you want to request a *feature* or report a *bug*?**

*bug*

**What is the current behavior?**

unexpected **grey splash** caused by `onMouseDown` on real ios device's browsers, both safari and chrome.
[Please checkout this **grey splash** issue first](https://github.com/mui-org/material-ui/issues/11154)
I did some trial later, and find out that vanilla html `<button onmousedown="">btn</button>` doesn't cause **grey splash**, but with React's jsx `<button onMouseDown={()=>{}}></button>`, there is a noticeable **grey splash**

**What is the expected behavior?**

who wants a splash?

**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**

"react": "^16.3.2"

**Comment by gaearon** [comment]
For future reference, the reproducing example is this:

```js
import React from "react";
import ReactDOM from "react-dom";

function Index() {
  return <button onMouseDown={() => {}}>Wtf</button>;
}

ReactDOM.render(<Index />, document.querySelector("#root"));
```

https://codesandbox.io/s/p3npm3o6j7

The splashes the author is referring to occur not when you click on the button, but when you click *on the body itself* several times in a quick succession.

**Comment by gaearon** [comment]
The problem is not specific to React; this also reproduces it:

```js
document.body.addEventListener(
  "mousedown",
  function() {}
);
```

**Comment by gaearon** [comment]
Here is what Polymer does:

https://github.com/Polymer/polymer/blob/59405b57b3f7f3732b7f1db07ef1a76b1871814e/lib/utils/gestures.js#L92-L93

https://github.com/Polymer/polymer/blob/59405b57b3f7f3732b7f1db07ef1a76b1871814e/lib/utils/gestures.js#L210

https://github.com/Polymer/polymer/blob/59405b57b3f7f3732b7f1db07ef1a76b1871814e/lib/utils/gestures.js#L545-L547

**Comment by gaearon** [comment]
Hmm.. there was clearly some issue with how I was testing this. It appears that CodeSandbox preview mode is buggy. I'll need to look at it again — for now previous conclusions might be wrong.

**Comment by gaearon** [comment]
I can't seem to reproduce this in a vanilla (non-CRA) template: https://codesandbox.io/s/n50krjw5v4

Any chance this is caused by some DEV-only code that CRA pulls in?

**Comment by gaearon** [comment]
Apparently removing 

```html
<!DOCTYPE html>
```

"fixes" this.

I have *no* idea why. But it does seem like a browser bug to me.

**Comment by philipp-spiess** [comment]
@gaearon You beat me to it 😬 

Here's another workaround I found 🤔 :

```css
html, body {
  width: 100%;
  height: 100%;
}
```

**Comment by philipp-spiess** [comment]
I don't think that #2043 would be a fix for this issue. If you mount your React app into a container that is big enough to fill the whole screen (which most React apps probably are) and then add the click listener to that container, the same issue persists: https://codesandbox.io/s/o4x23pr8m6

What's different with the non-`document` listeners though:

1. The behavior is always reproducible, no matter if we add `<!DOCTYPE html>` or not.
2. We can use `-webkit-tap-highlight-color: transparent;` to remove it.

I've filed a WebKit issue for the behavior we're seeing with `document` as it seems very inconsistent: https://bugs.webkit.org/show_bug.cgi?id=188362
